### ISSUE TITLE
Pay for Slack

### ISSUE THREAD

**Issue by geeknoid** [istio/istio#191]
We should set up a paying account with Slack so that our messages are archived for ever. As it is right now, stuff falls off the end of the world after a while.

**Comment by ayj** [comment]
[Slack for Teams](https://istio.slack.com/pricing/slack-for-teams) is billed per user. There is an [enterprise option](https://istio.slack.com/pricing/enterprise-grid) that might be more appropriate.

**Comment by swade1987** [comment]
Would also be useful to allow users to have a place to ask the dev team questions

**Comment by ZackButcher** [comment]
@swade1987 we have istio.slack.com which we intend to use for users interacting with devs (this issue is for upgrading it to some paid version).

**Comment by swade1987** [comment]
@ZackButcher sure but right now users have no way of interacting with devs using Slack. Is this on the roadmap?

**Comment by josephjacks** [comment]
@geeknoid Ideally, Istio doesn't have to pay for anything. :-)

https://www.cncf.io/blog/2017/03/01/slack-gives-back-k8s-cncf-community/

**Comment by ZackButcher** [comment]
@swade1987 Yes; AFAIK the plan is to open it up when we start making noise about Istio.

**Comment by zinuga** [comment]
@geeknoid I checked with K8s and they went on free slack for a very long time. may be we start with that? Anyways, seems like searchability beyond last 10K messages is a good to have.

**Comment by geeknoid** [comment]
We needed to pay for Slack for 2 reasons:

- Retention beyond 10K
- Ability to have groups so we can control who can join individual channels

Since we've decided not to offer Slack to end-users, we no longer need the
ability to control permissions, leaving us just the retention thing.

At this point, I really want to push design discussions to the istio-dev
mailing list instead. So Slack ends up just being for short-lived
spur-of-the-moment discussions, similar to what you'd do in a hallway. So I
think it's reasonable for that to be limited to 10K messages.

So yeah, let's just punt on this.
